lib/tessel/deploy.js

```diff
@@ -205,49 +211,98 @@ actions.sendBundle = function(tessel, filepath, opts) {
 
 actions.tarBundle = function(opts) {
   var target = opts.target || process.cwd();
+  var pack = tar.Pack();
+  var buffers = [];
+
+  if (opts.slim) {
+    console.log('Running slim...');
+    return new Promise(function(resolve, reject) {
+      var b = browserify(opts.resolvedEntryPoint, {
+        node: true,
+        ignoreMissing: true
+      });
+      b.transform(uglify);
```

Comment:
It is a Browserify transform for minifying the code being bundled. [Reference](https://www.npmjs.com/package/uglifyify). I added it as another step to reduce the size of the tarball.